core/src/com/mygdx/game/GameEngine.java

```diff
@@ -146,7 +146,13 @@ public float addSingleResult(LocalRayResult rayResult, boolean normalInWorldSpac
 	private Array<GameModel> dynamicModels = new Array<GameModel>();
 	private Bits visibleLayers = new Bits();
 
+	// FIXME
+	// Ugly hack to access the engine from anywhere
+	public static GameEngine engine;
+
```

Comment:
Yeah, I did it like that just because I was focused on dog AI and wanted it to work somehow.
A singleton would be probably better if we don't want to pass the engine reference everywhere is needed.